### ISSUE TITLE
Terminal a11y help menu includes wrong keybinding

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#175130]
With `editor.accessibilitySupport:true`, the help menu says that the keybinding to `Run recent command` is `Ctrl+Alt+R` on windows or `Command+Option+R` on macos. 

It should instead correctly reflect the keybinding used for this when context: `Ctrl+R` and `Command+R`. 

This is probably because the terminal is not focused when the help menu is shown

**Comment by amunger** [comment]
I have `"editor.accessibilitySupport": "on"`, but I can't seem to get the context `accessibilityModeEnabled` to be true, ctrl+alt+R is always the correct keybinding. Is there another setting I'm missing?

And after reloading, I no longer see the command in the help menu at all (shift+tab to focus on terminal buffer doesn't help)
![image](https://user-images.githubusercontent.com/2019016/227000680-bf5e6692-2736-4007-af80-8c884791b7ae.png)

**Comment by meganrogge** [comment]
You're in a terminal without shell integration. Open a normal pwsh terminal and you should see this 

![Screenshot 2023-03-22 at 11 41 30 AM](https://user-images.githubusercontent.com/29464607/227005525-8ccb6ce0-8c07-43ce-a0a3-f7d3aef15ae7.png)

**Comment by meganrogge** [comment]
I had changed my `when` clause 🤦🏼‍♀️

**Comment by meganrogge** [comment]
moving the `TerminalContextKeys.focus` from both `when` conditions to the `precondition` fixes it, though not sure why

![Screenshot 2023-03-22 at 1 59 48 PM](https://user-images.githubusercontent.com/29464607/227036972-92fef4be-f40a-4f89-b438-cdc128ca0c6e.png)

**Comment by meganrogge** [comment]
terminal focus context key is undefined, perhaps because of this scoped context key service?
![Screenshot 2023-03-22 at 2 09 23 PM](https://user-images.githubusercontent.com/29464607/227038937-b7f477af-c6ec-45da-a727-8d911fb45225.png)

https://github.com/microsoft/vscode/blob/2e954c8b2d71db71362425c8ba07840aa4642f41/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L446

**Comment by meganrogge** [comment]
this is `true` at that same location, so perhaps we should update the terminal focus context key to have that value

https://github.com/microsoft/vscode/blob/2e954c8b2d71db71362425c8ba07840aa4642f41/src/vs/workbench/contrib/terminal/browser/terminal.ts#L512

**Comment by meganrogge** [comment]
So it turns out that `lookupKeybinding` just returns whichever is first - regardless of the `when` context. 


https://github.com/microsoft/vscode/blob/a0248293d5f7820712646dbd84dd22f886ac3224/src/vs/platform/keybinding/common/keybinding.ts#L70-L80

**Comment by meganrogge** [comment]
@alexdima here is the hack that I did to fix this - pls lmk if there's a better way

the relevant code

https://github.com/microsoft/vscode/blob/7b1e98a64b2b4efecbafe74c6534e6ac113e25c3/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L320-L338

https://github.com/microsoft/vscode/blob/a0248293d5f7820712646dbd84dd22f886ac3224/src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibilityHelp.ts#L54-L66

**Comment by alexdima** [comment]
@meganrogge If you pass in the context as the second argument, then `lookupKeybinding` will return the keybinding which matches the passed in context:

```
lookupKeybinding(commandId: string, context?: IContextKeyService): ResolvedKeybinding | undefined; 
```

**Comment by meganrogge** [comment]
when I tried that, it didn't work for me

**Comment by alexdima** [comment]
I see the argument being used [here](https://github.com/microsoft/vscode/blob/07aec54e2ebac3de9dbd6c0ef600fb9dc3cbba6b/src/vs/platform/keybinding/common/abstractKeybindingService.ts#L128). If you are able to distill reproducible steps, I'm happy to have a look.

**Comment by amunger** [comment]
I'm still seeing the keybindings not honoring the when clause - ctrl+alt+R is always matched, and I confirmed that `accessibilityModeEnabled: true`

![terminalCommands](https://user-images.githubusercontent.com/2019016/234943595-d813670b-21f5-4c47-a988-282500dfe7b0.gif)


keybinding logs

```
2023-04-27 10:29:41.569 [info] [KeybindingService]: | Resolving ctrl+alt+R
2023-04-27 10:29:41.569 [info] [KeybindingService]: \ From 2 keybinding entries, matched workbench.action.terminal.sendSequence, when: accessibilityModeEnabled && terminalFocus, source: built-in.
2023-04-27 10:29:41.570 [info] [KeybindingService]: / Received  keydown event - modifiers: [ctrl,alt], code: KeyR, keyCode: 82, key: r
2023-04-27 10:29:41.571 [info] [KeybindingService]: | Converted keydown event - modifiers: [ctrl,alt], code: KeyR, keyCode: 48 ('R')
2023-04-27 10:29:41.571 [info] [KeybindingService]: | Resolving ctrl+alt+R
2023-04-27 10:29:41.571 [info] [KeybindingService]: \ From 2 keybinding entries, matched workbench.action.terminal.sendSequence, when: accessibilityModeEnabled && terminalFocus, source: built-in.
2023-04-27 10:29:41.572 [info] [KeybindingService]: + Invoking command workbench.action.terminal.sendSequence.
2023-04-27 10:29:41.800 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-04-27 10:29:41.801 [info] [KeybindingService]: | Resolving ctrl+alt+R
2023-04-27 10:29:41.802 [info] [KeybindingService]: \ From 2 keybinding entries, matched workbench.action.terminal.sendSequence, when: accessibilityModeEnabled && terminalFocus, source: built-in.
```

**Comment by meganrogge** [comment]
@alexdima could you pls have a look? here are the logs, not sure what I'm doing wrong here

```
KeybindingService]: | Resolving ctrl+R
2023-04-27 11:19:09.717 [info] [KeybindingService]: \ From 4 keybinding entries, matched workbench.action.terminal.runRecentCommand, when: accessibilityModeEnabled && terminalFocus && terminalHasBeenCreated || accessibilityModeEnabled && terminalFocus && terminalProcessSupported, source: built-in.
2023-04-27 11:19:09.780 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-04-27 11:19:09.781 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2023-04-27 11:19:09.781 [info] [KeybindingService]: + Ignoring single modifier ctrl due to it being pressed together with other keys.
```

**Comment by alexdima** [comment]
@ulugbekna Please investigate

**Comment by ulugbekna** [comment]
The keybinding lookup does respect the context/when-clauses only when at least one of the keybindings' when clause evaluates to true. In this particular case, as far as I see from debugging [1], the when clauses of both keybindings evaluate to false (`terminalFocus` is undefined in the context), so the lookup simply picks the last keybinding added for this command, which happens to be `ctrl+alt+r` - [source](https://github.com/microsoft/vscode/blob/d7da8fc2c964595c25ecf749345df6e2079a2b9e/src/vs/platform/keybinding/common/keybindingResolver.ts#L293-L300) 

<img width="414" alt="image" src="https://user-images.githubusercontent.com/16353531/235129656-d5477f01-fcb5-4b89-9d5f-cbb0efb869cb.png">

[1] For debugging, I looked up the keybinding for "run recent command..." in the command palette, not the terminal accessibility help because I didn't see this command there (also the help appears for me as text rather than hover) 

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/16353531/235132718-e2a13b53-e405-4da7-abd7-ff9ac34a1232.png">

note: this is in OSS ^

-----

I think the fix would be to either have `terminalFocus` set to true in the context passed to the lookup function or we change how we break ties among keybindings where when-clauses evaluate to false (a missing context key should have higher priority that a context key that evaluates to false?)

**Comment by meganrogge** [comment]
Thanks for investigating @ulugbekna. 

> we change how we break ties among keybindings where when-clauses evaluate to false (a missing context key should have higher priority that a context key that evaluates to false)

I think this makes sense

**Comment by meganrogge** [comment]
@ulugbekna could you pls work on this soon? It's pretty bad that we currently list the wrong keybinding in our accessibility dialogs.

**Comment by ulugbekna** [comment]
@meganrogge I hope to get to this in October iteratoin
